### ISSUE TITLE
Integration authorization does not support http/2

### ISSUE THREAD

**Issue by therool** [magento/magento2#20166]
### Preconditions (*)

1. Magento 2.3 but I guess on other versions also
1. The magento 2 server needs to run on http/2, for example apache with https://httpd.apache.org/docs/2.4/howto/http2.html
1. Issue is with "Magento\Integration" module specifically with the ["Magento\Integration\Model\OauthService"](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Integration/Model/OauthService.php) as it uses "Magento\Framework\HTTP\ZendClient" which is not http/2 compliant

### Steps to reproduce (*)

1. Add any integration
2. Press authorize

### Expected result (*)

1. Authorization process continues

### Actual result (*)

1. Will see an error message ["The attempt to post data to consumer failed due to an unexpected error. Please try again later."](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Integration/Model/OauthService.php#L251)

### Additional information

The issue is that the ["Magento\Integration\Model\OauthService"](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Integration/Model/OauthService.php#L87) uses ["Magento\Framework\HTTP\ZendClient"](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/HTTP/ZendClient.php#L40) which is not http/2

Backtrace:
#0 Zend_Http_Response::extractHeaders() called at [{magento2-root}/vendor/magento/zendframework1/library/Zend/Http/Response.php:696](https://github.com/magento/zf1/blob/master/library/Zend/Http/Response.php#L696)
#1 Zend_Http_Response::fromString() called at [{magento2-root}/vendor/magento/zendframework1/library/Zend/Http/Client.php:1110](https://github.com/magento/zf1/blob/master/library/Zend/Http/Client.php#L1110)
#2 Zend_Http_Client->request() called at [{magento2-root}/vendor/magento/framework/HTTP/ZendClient.php:52](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/HTTP/ZendClient.php#L52)
#3 Magento\Framework\HTTP\ZendClient->request() called at [{magento2-root}/vendor/magento/module-integration/Model/OauthService.php:243](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Integration/Model/OauthService.php#L243)

The faulty line under  [{magento2-root}/vendor/magento/zendframework1/library/Zend/Http/Response.php:517](https://github.com/magento/zf1/blob/master/library/Zend/Http/Response.php#L517) which supports only http/1.1 and not http/2.

In short, use different http client for ["Magento\Integration\Model\OauthService"](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Integration/Model/OauthService.php#L87) or make the [{magento2-root}/vendor/magento/zendframework1/library/Zend/Http/Response.php:517](https://github.com/magento/zf1/blob/master/library/Zend/Http/Response.php#L517) http/2 compliant

**Comment by magento-engcom-team** [comment]
Hi @therool. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@therool do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
HI @therool thank you for you report, magento  not support HTTP/2 you can report here -> https://github.com/magento/community-features/issues

**Comment by therool** [comment]
Hello @engcom-backlog-nazar,
I have an client who is using magento 2.3 with HTTP/2, I need to advise him to update his server to use HTTP/1 instead and ignore his issues regarding HTTP/2?

Regards,
Ricards Z

**Comment by ghost** [comment]
@therool I thinks yes, support for http/2 will be available soon.
